modules/psap-installing-node-feature-discovery-operator.adoc

```diff
@@ -0,0 +1,144 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/psap-node-feature-discovery-operator.adoc
+
+[id="installing-the-node-feature-discovery-operator_{context}"]
+= Installing the Node Feature Discovery Operator
+
+The Node Feature Discovery (NFD) Operator manages detection of hardware features and configuration in a kubernetes cluster. The Operator orchestrates all resources needed to run the Node-Feature-Discovery daemon set As a cluster administrator, you can install the NFD Operator using the {product-title} CLI or the web console.
+
+[id="install-operator-cli_{context}"]
+== Installing the NFD Operator using the CLI
+
+As a cluster administrator, you can install the Operator using the CLI.
+
+.Prerequisites
+
+* An {product-title} cluster
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create a namespace for the Node Feature Discovery Operator.
+
+.. Create the following `Namespace` custom resource (CR) that defines the `openshift-nfd-operator` namespace, and then save the YAML in the `nfd-namespace.yaml` file:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openshift-nfd-operator
+----
+
+.. Create the namespace by running the following command:
++
+[source,terminal]
+----
+$ oc create -f nfd-namespace.yaml
+----
+
+. Install the NFD Operator in the namespace you created in the previous step by creating the following objects:
+
+.. Create the following `OperatorGroup` CR and save the YAML in the `nfd-operatorgroup.yaml` file:
++
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  generateName: openshift-nfd-
+  name: openshift-nfd
+  namespace: openshift-nfd-operator
+spec:
+  targetNamespaces:
+  - openshift-nfd-operator
+
+----
+
+.. Create the `OperatorGroup` CR by running the following command:
++
+[source,terminal]
+----
+$ oc create -f nfd-operatorgroup.yaml
+----
+
+.. Run the following command to get the `channel` value required for the next step.
++
+[source,terminal]
+----
+$ oc get packagemanifest openshift-nfd-operator -n openshift-marketplace -o jsonpath='{.status.defaultChannel}'
+----
++
+.Example output
+[source,terminal]
+----
+nfd.v4.8.0
+----
+
+.. Create the following `Subscription` CR and save the YAML in the `nfd-sub.yaml` file:
++
+.Example Subscription
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: nfd
+  namespace: openshift-nfd-operator
+spec:
+  channel: nfd.v4.8.0
+  installPlanApproval: Automatic
+  name: nfd
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+----
+
+.. Create the subscription object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f nfd-sub.yaml
+----
+
+.. Change to the `openshift-nfd-operator` project:
++
+[source,terminal]
+----
+$ oc project openshift-nfd-operator
+----
```

Comment:
Can a step be added here to validate that the operator deployment succeeds ?

Reply:
Could be something like 
```bash
$ oc get pods
NAME                            READY   STATUS    RESTARTS   AGE
nfd-master-c92jg                1/1     Running   0          48s
nfd-master-f7gd9                1/1     Running   0          48s
nfd-master-jdg8t                1/1     Running   0          48s
nfd-operator-5b945f7ddd-kt4mm   1/1     Running   0          79s
nfd-worker-6wh9m                1/1     Running   0          47s
nfd-worker-crchk                1/1     Running   0          47s
nfd-worker-mgl7g                1/1     Running   0          47s
```

Reply:
@ahardin-rh ^